### PR TITLE
QA and CI: Format code using ruff. Validate using ruff and mypy.

### PR DESCRIPTION
**Pull request by amotl**
## About
- Run the ruff code formatter and linter on the code base, and the mypy type checker.
- Invoke the "lint" job also on CI.

## Rationale
Before releasing 1.0.0, we would like to get the shape of the repository in order. This patch is one in a series to support that.

## Details
No other functional changes inside.

## References
- GH-659

### CODE REVIEW

**Review by matriv — APPROVED** [review]

File: src/crate/client/cursor.py [hunk 1]
```diff
@@ -306,12 +300,17 @@ def _to_datetime_with_tz(value: t.Optional[float]) -> t.Optional[datetime]:
     @staticmethod
     def _timezone_from_utc_offset(tz) -> timezone:
         """
-        Convert UTC offset in string format (e.g. `+0530`) into `datetime.timezone` object.
+        UTC offset in string format (e.g. `+0530`) to `datetime.timezone`.
         """
-        assert len(tz) == 5, f"Time zone '{tz}' is given in invalid UTC offset format"
+        # TODO: Remove use of `assert`. Better use exceptions?
```
Comment (kneth) [inline src/crate/client/cursor.py:305]: Only if you will catch the exception where you call `_timezone_from_utc_offset`

**Review by kneth — APPROVED** [review]
Reformatting always gives you huge PRs.

File: src/crate/client/cursor.py (hunk 1 above)
Comment (amotl) [reply src/crate/client/cursor.py:305]: On this spot, I am referring to the fact that the ruff linter discourages the use of `assert` statements in code:
```
S101 Use of `assert` detected
```

That's why it needs to be suppressed now, using `# noqa: S101`.

My proposal would be to rewrite the `assert` statements to constructs like that:
```python
if len(tz) != 5:
    raise ValueError(f"Time zone '{tz}' is given in invalid UTC offset format")
```

I think that would be the correct approach to modernize the code base. wdyt?

Comment (kneth) [reply src/crate/client/cursor.py:305]: Now it makes sense, and I agree that raising an exception is a better option.

**Pull request merged**